src/pkg/exporter/cache.go

```diff
@@ -0,0 +1,88 @@
+package exporter
+
+import (
+	"sync"
+	"time"
+)
+
+var c *cache
+
+type cachedValue struct {
+	Value      interface{}
+	Expiration int64
+}
+
+type cache struct {
+	CacheTime int64
+	store     map[string]cachedValue
+	*sync.RWMutex
+}
+
+// CacheGet get a value from cache
+func CacheGet(key string) (value interface{}, ok bool) {
+	c.RLock()
+	v, ok := c.store[key]
+	c.RUnlock()
+	if !ok {
+		return nil, false
+	}
+	if time.Now().Unix() > v.Expiration {
+		c.Lock()
+		delete(c.store, key)
+		c.Unlock()
+		return nil, false
+	}
+	return v.Value, true
+}
+
+// CachePut put a value to cache with key
+func CachePut(key, value interface{}) {
+	c.Lock()
+	defer c.Unlock()
+	c.store[key.(string)] = cachedValue{
+		Value:      value,
+		Expiration: time.Now().Unix() + c.CacheTime,
+	}
+}
+
+// CacheDelete delete a key from cache
+func CacheDelete(key string) {
+	c.Lock()
+	defer c.Unlock()
+	delete(c.store, key)
+}
+
+// StartCacheCleaner start a cache clean job
+func StartCacheCleaner() {
```

Comment:
why not call CacheClean... this func is a verb..

Reply:
CacheClean can not fully describe this function

Reply:
It's not an async func, you are not starting any job or goroutine.  I really can't see why we have to add `Start`.

Reply:
check  the `CacheInit` func